internal/services/user/user_test.go

```diff
@@ -172,121 +150,107 @@ func TestGetGroupByGID(t *testing.T) {
 			// We don't care about gpasswd output here as it's already covered in the db unit tests.
 			_ = localgroupstestutils.SetupGPasswdMock(t, filepath.Join("testdata", "empty.group"))
 
-			client := newNSSClient(t, tc.sourceDB, false)
+			client := newUserServiceClient(t, tc.dbFile)
 
-			got, err := client.GetGroupByGID(context.Background(), &authd.GetByIDRequest{Id: tc.gid})
+			got, err := client.GetGroupByGID(context.Background(), &authd.GetGroupByIDRequest{Id: tc.gid})
 			requireExpectedResult(t, "GetGroupByGID", got, err, tc.wantErr, tc.wantErrNotExists)
 		})
 	}
 }
 
-func TestGetGroupEntries(t *testing.T) {
-	tests := map[string]struct {
-		sourceDB string
-
-		wantErr bool
-	}{
-		"Return_all_groups": {},
-		"Return_no_groups":  {sourceDB: "empty.db.yaml"},
-	}
-	for name, tc := range tests {
-		t.Run(name, func(t *testing.T) {
-			// We don't care about gpasswd output here as it's already covered in the db unit tests.
-			_ = localgroupstestutils.SetupGPasswdMock(t, filepath.Join("testdata", "empty.group"))
-
-			client := newNSSClient(t, tc.sourceDB, false)
-
-			got, err := client.GetGroupEntries(context.Background(), &authd.Empty{})
-			requireExpectedEntriesResult(t, "GetGroupEntries", got.GetEntries(), err, tc.wantErr)
-		})
-	}
+func TestMockgpasswd(t *testing.T) {
+	localgroupstestutils.Mockgpasswd(t)
```

Comment:
This is technically a helper function, so I think it's better for us to put it after the unit tests.